src/octotree.js

```diff
@@ -24,9 +25,26 @@ $(document).ready(() => {
       if (event.target === window) layoutChanged();
     });
 
-    $toggler.click(toggleSidebarAndSave);
+    $document.click((event) => !isSidebarPinned() && isOutsideSidebar(event.target) && toggleSidebar(false));
+
+    $pinner.click(togglePin);
+
+    $sidebar.on(
+      `keyup mouseleave mousemove ${EVENT.PIN}`,
+      debounce({
+        callback: () => toggleSidebar(false),
+        // Mousemove handles the edge case in which mouseleave and then immediately mouseenter
+        cancelIf: (event) => isSidebarPinned() || event.type === 'mousemove',
+        delay: DEFAULTS.SHOWTIME
```

Comment:
Shouldn't use DEFAULTS for non-storing var. Just introduce a constant, i.e. TREE_HIDING_DELAY